doc/specs/#120,121,119-Update-Uninstall-List.md

```diff
@@ -0,0 +1,230 @@
+    
+```HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\{3740BD44-B58D-321A-AFC0-6D3D4556DD6C}]```
+
+For inno, wix, nullsoft, and exe, the SystemAppId should be a string that is located in either of the Uninstall keys above. 
+
+### Upgrade
+
+The **upgrade** command is designed to update one or more applications.  The **upgrade** command when executed with no parameters or with --all will update all packages that have updates available.
+
+
+## Usage
+
+`winget upgrade [[-q] \<query>] [\<options>]`
+
+![show command]()
+
+**Upgrade** when executed by itself will upgrade all apps ready for updates by the Windows Package Manager.
+
+
+| Argument  | Description |
+|--------------|-------------|
+| **-q,--query** |  The query used to search for an application. |
+| **-?, --help** |  Gets additional help on this command. |
+
+## Options
+
+The following options are available.
+
+| Option  | Description |
+|--------------|-------------|
+| **--all**         |  update all applications that are installed and have updates available.|
+| **--id**         |  Filter results by ID. |
+| **--name**   |      Filter results by name. |
+| **--moniker**   |  Filter results by application moniker. |
+| **--tag** |     Filter results by tag
+| **--command**  | Filter results by command
+| **-s,--source** |   Find the application using the specified [source](source.md). |
+| **-e,--exact**     | Find the application using exact match. |
+
+
+*Upgrade app when no update is available*  
+If the user attempts to update an app that there is no known update for, the package manager will issue an error.
+"There is no update available from the community repository.  If you need to reinstall or repair the app, try ```winget repair <appname>```.
+
+
+
+*Depnedencies*
+Handling dependencies will be a challenge once Windows Package Manager supports dependencies.  Dependencies are not covered as part of this spec.
+
+
+
+### Uninstall
+
+The **uninstall** command is designed to uninstall an application.   
+
+## Usage
+
+`winget uninstall [[-q] \<query>] [\<options>]`
+
+![show command]()
+
+
+| Argument  | Description |
+|--------------|-------------|
+| **-q,--query** |  The query used to search for an application. |
+| **-?, --help** |  Gets additional help on this command. |
+
+## Options
+
+The following options are available.
+
+| Option  | Description |
+|--------------|-------------|
+| **--id**         |  Filter results by ID. |
+| **--name**   |      Filter results by name. |
+| **--moniker**   |  Filter results by application moniker. |
+| **--tag** |     Filter results by tag
+| **--command**  | Filter results by command
+| **-s,--source** |   Find the application using the specified [source](source.md). |
+| **-e,--exact**     | Find the application using exact match. |
+
+
+
+
+
+
+
+
+
+
+
+
+
+[comment]: # Outline the design of the solution. Feel free to include ASCII-art diagrams, etc.
+
+## UI/UX Design
```

Comment:
A suggestion for what could be discussed here: The impact on users that have not used a package manager, will they naturally use `update` instead of `upgrade` (or sometimes have issues with distinguishing these two)?  And on the other hand, if `update` was an alias for `upgrade`, how can similar functionality of `upgrade` be used with the alias and vice versa? (For example: https://github.com/microsoft/winget-cli/issues/120#issuecomment-631945047) It would be really nice to see what is the plan for dealing with this kind of complexity (`update` vs. `upgrade`) from a user standpoint.

Reply:
I prefer "update'.  Upgrade has other semantics (at least to me), and update fits more cleanly into what is happening.

Reply:
@JohnMcPMS If you decide to go with `update`, then I would suggest to make `upgrade` alias of `update` to make the life easier for users familiar with other package managers that use `upgrade` for updating packages.

Reply:
Brew user here and every time I want to update an app, I have to look up help to see if I need to `update` or `upgrade`. One is making sure Brew is up to date, the other is for updating the app. If you trigger update without supplying app identifier then Brew will attempt to update every installed packages. Which I'm not a fan of. 

Personally, I'm against making alias between `update` and `upgrade`. Having just one version of them at the start should reduce confusion on which one to use for newcomers. Instead of doing this, I'd suggest WinGet to only offer just one.

- `winget update` - displaying help for update. 
- `winget update <package_id>` - updating provided package.
- `winget update winget` - updating WinGet itself.
- `winget update -a|--all` - updating every installed packages.

Reply:
I think winget should use keywork like apt of Debian.

- Update: for refresh the catalog
- Upgrade: for actual installation operation of new version

This way we could remove the refresh frequency as winget user does not expect winget run in background as a CLI tool usually run and exit